File: MyTennisPartner.Models/ViewModels/MatchDetailsViewModel.cs
using MyTennisPartner.Models.Enums;
using System;
using System.Collections.Generic;

namespace MyTennisPartner.Models.ViewModels
{
    /// <summary>
    /// class to hold basic sumamry info about matches, lines, and players
    /// </summary>
    public class MatchDetailsViewModel
    {
        public MatchDetailsViewModel(List<MatchSummaryViewModel> matches, List<LineViewModel> lines, List<PlayerViewModel> players)
        {
            Matches = matches;
            Lines = lines;
            Players = players;
        }

        public List<MatchSummaryViewModel> Matches { get; }
        public List<LineViewModel> Lines { get; }
        public List<PlayerViewModel> Players { get; }
    }
}

File: MyTennisPartner.Web/ClientApp/models/viewmodels/MatchDetailsViewModel.ts
import LeagueSummaryViewModel from "./LeagueSummaryViewModel";
import MatchSummaryViewModel from "./MatchSummaryViewModel";
import LineViewModel from "./LineViewModel";
import PlayerViewModel from "./PlayerViewModel";

export default class MatchDetailsViewModel {
    constructor() {
        this.matches = [] as MatchSummaryViewModel[];
        this.lines = [] as LineViewModel[];
        this.players = [] as PlayerViewModel[];
    }
    matches: MatchSummaryViewModel[];
    lines: LineViewModel[];
    players: PlayerViewModel[];
}